sensor/common/networkflow/metrics/metrics.go

```diff
@@ -1,122 +1,221 @@
-		activeFlowsTotal,
-		activeEndpointsTotal,
+		activeFlowsCurrent,
+		activeEndpointsCurrent,
+		PurgerEvents,
+		ActiveEndpointsPurgerDuration,
+		NumUpdatedConnectionsEndpoints,
+
+		// Other
+		NetworkEntityFlowCounter, // flow directions and graph entities
+		HostProcessesEvents,      // plop
+		HostProcessesEnrichmentEvents,
 	)
 }
 
+const (
+	hostConnectionsPrefix = "host_connections_"
+	netFlowManagerPrefix  = "network_flow_manager_"
+)
+
 // Metrics for network flows
 var (
-	NetworkFlowsPerNodeByType = prometheus.NewCounterVec(prometheus.CounterOpts{
+	// A networkConnectionInfo message arrives from collector
+
+	// NetworkConnectionInfoMessagesRcvd - 1. Collector sends NetworkConnection Info messages where each contains endpoints and connections
+	NetworkConnectionInfoMessagesRcvd = prometheus.NewCounterVec(prometheus.CounterOpts{
 		Namespace: metrics.PrometheusNamespace,
 		Subsystem: metrics.SensorSubsystem.String(),
-		Name:      "network_flow_total_per_node",
-		Help:      "Total number of network flows received for a specific node",
-	}, []string{"Hostname", "Type", "Protocol"})
-	ContainerEndpointsPerNode = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:      hostConnectionsPrefix + "msgs_received_per_node_total",
+		Help:      "Total number of messages containing network flows received from Collector for a specific node",
+	}, []string{"Hostname"})
+	// NumUpdated - 2. Out of newly arrived endpoints and connections, only selected need an update
+	NumUpdated = prometheus.NewGaugeVec(prometheus.GaugeOpts{
 		Namespace: metrics.PrometheusNamespace,
 		Subsystem: metrics.SensorSubsystem.String(),
-		Name:      "network_endpoints_total_per_node",
-		Help:      "Total number of container endpoint updates received for a specific node",
-	}, []string{"Hostname", "Protocol"})
-	NetworkFlowMessagesPerNode = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:      hostConnectionsPrefix + "num_updates",
+		Help:      "Current number of network endpoints or connections being updated in the message from Collector received for a specific node",
+	}, []string{"Hostname", "Type"})
+	// HostConnectionsOperations - 3a. Out of the updates, only some result in adding the connection/endpoint to the map
+	HostConnectionsOperations = prometheus.NewCounterVec(prometheus.CounterOpts{
 		Namespace: metrics.PrometheusNamespace,
 		Subsystem: metrics.SensorSubsystem.String(),
-		Name:      "network_flow_msgs_received_per_node",
-		Help:      "Total number of network flows received for a specific node",
-	}, []string{"Hostname"})
-	ContainerIDMisses = prometheus.NewCounter(prometheus.CounterOpts{
+		Name:      hostConnectionsPrefix + "operations_total",
+		Help:      "Total number of flows/endpoints added/removed in the host connections maps",
+	}, []string{"op", "object"})
+	// NumHostConnections - 3b. how many entries does Sensor hold in the hostConnections map?
+	NumHostConnections = prometheus.NewGauge(prometheus.GaugeOpts{
```

Comment:
Couldn't we get this information from `HostConnectionsOperations`?

Reply:
Yes, we could, it looks like a duplication here. However, it is much easier to query that in Grafana instead of doing subtraction of two metric selectors.
I have a stacked PR with Grafana dashboard - I will see whether we can subtract the adds and removes and get this number on the graph.

Reply:
This is now removed